lagranlucha-client/Assets/_Main/Scripts/Game/UI/OrderManager.cs

```diff
@@ -0,0 +1,227 @@
+
+        #region PROPERTIES
+
+        public bool OrderComplete { get => completedHamburguer && completedFries && completedDrink; }
+        public bool HaveTimeRemaining { get => timer.IntValue > default(int); }
+        public float ThisRoundHamburguersScore { get; private set; }
+        public float ThisRoundFriesScore { get; private set; }
+        public float ThisRoundDrinksScore { get; private set; }
+        public float ThisRoundTotalScore { get => ThisRoundHamburguersScore + ThisRoundFriesScore + ThisRoundDrinksScore; }
+
+        #endregion
+
+        #region BEHAVIORS
+
+        private void Start()
+        {
+            orderUI.onTrayCleaned += OrderSent;
+            gameManager.onRoundStart += StartRound;
+        }
+
+        private void OnDestroy()
+        {
+            orderUI.onTrayCleaned -= OrderSent;
+            gameManager.onRoundStart -= StartRound;
+        }
+
+        private void StartRound(int round)
+        {
+            ThisRoundHamburguersScore = ThisRoundFriesScore = ThisRoundDrinksScore = default(int);
+            Invoke(nameof(NewOrder), DelayRoundStart);
+        }
+
+        private void NewOrder()
+        {
+            order = roundOrder.NewOrder(gameManager.CurrentRound);
+            order.GetRandomElements();
+            currentIngredientArrow.SetActive(order.HaveHamburger);
+            completedHamburguer = !order.HaveHamburger;
+            completedFries = !order.HaveFries;
+            completedDrink = !order.HaveDrink;
+            hamburguerStep = default(int);
+            SetGraphics();
+            shownName.text = order.OrderName;
+            onNewOrderStart?.Invoke(order);
+        }
+
+        public void SendIngredient(Ingredient ingredient)
+        {
+            if (ingredient == GetFriesIngredient())
+            {
+                onFriesIngredientSubmit(ingredient);
+                friesIngredient.Hide();
+                completedFries = true;
+            }
+            else if (ingredient == GetDrinkIngredient())
+            {
+                onDrinkIngredientSubmit(ingredient);
+                drinkIngredient.Hide();
+                completedDrink = true;
+            }
+            else if (ingredient == GetHamburguerIngredient(hamburguerStep))
+            {
+                onHamburguerIngredientSubmit(ingredient);
+                hamburguerIngredients[hamburguerStep].Hide();
+                hamburguerStep++;
+                if (hamburguerStep >= order.HamburgerIngredientsCount)
+                    completedHamburguer = true;
+                else
+                    hamburguerIngredients[hamburguerStep].SetAsCurrentIngredient();
+            }
+            else
+            {
+                if (ingredient.IsHamburguerIngredient && !completedHamburguer)
+                    ResetBurguer();
+
+                ShowFail();
+            }
+
+            if (OrderComplete)
+                CompletedOrder();
+        }
+
+        private void ShowFail() { }
```

Comment:
This is empty.

Reply:
There will be a sound or animation but it still not defined yet

Reply:
Throw an Not defined exception or something like that when doing this.

Reply:
Done